File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.java
import java.util.Arrays;

class Solution {
    public int[] solution(String s) {
        int[] answer = new int[s.length()];

        for(int i=0; i<s.length(); i++){
            String str = s.substring(0,i);
            if(str.indexOf(s.charAt(i))==-1) {
                answer[i] = -1;
            }else {
                answer[i] = i-str.lastIndexOf(s.charAt(i));
            }
        }
        return answer;
    }
}

File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.js
function solution(s) {
    var answer = [];
    let arr = [];
    for(let i=0; i<s.length; i++) {
        if(!arr.includes(s[i])) answer.push(-1);
        else answer.push(i-arr.lastIndexOf(s[i]));
        arr.push(s[i]);
    }
    
    return answer;
}